compiler/ballerina-lang/src/main/java/io/ballerina/projects/JarLibrary.java

```diff
@@ -86,4 +86,21 @@ public Path path() {
     public Optional<String> groupId() {
         return Optional.ofNullable(groupId);
     }
+
+    @Override
+    public boolean equals(Object other) {
+
+        if (this == other) {
+            return true;
+        }
+
+        if (other == null || getClass() != other.getClass()) {
+            return false;
+        }
+
+        JarLibrary that = (JarLibrary) other;
+
+        return path().toFile().toString().equals(that.path().toFile().toString());
```

Comment:
we should compare `path().toFile().getName()`

Reply:
changed to path().toFile().getName()